### ISSUE TITLE
Bug: (fill in)

### ISSUE THREAD

**Issue by marekdon** [eslint/eslint#16829]
### Environment

Node version: 16.15.1
npm version: 8.12.11
Local ESLint version: 8.30.0
Global ESLint version: 8.30.0
Operating System: iOS

### What parser are you using?

@angular-eslint/template-parser

### What did you do?

<details>
<summary>Configuration</summary>

```
<!-- Paste your configuration here -->
```
</details>

```js
const baseCardholderNameSurnameRegexString = `^([\d\s'.-_\`a-z]{2,29}$)`;
...
...
const cardholderRegex = new RegExp(
    baseCardholderNameSurnameRegexString,
    'i'
  ),
```

### What did you expect to happen?

Rule should not suggest to remove slashes, as it will change the regex string and the output

### What actually happened?

Rule suggested that removal of slashes will not change the output, after slashes removal, Regex changed.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @marekdon, thanks for the issue!

This works as intended because `no-useless-escape` rule reports `\d` and `\s` ([Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgYmFzZUNhcmRob2xkZXJOYW1lU3VybmFtZVJlZ2V4U3RyaW5nID0gYF4oW1xcZFxccycuLV9cXGBhLXpdezIsMjl9JClgOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibm8tdXNlbGVzcy1lc2NhcGUiOlsiZXJyb3IiXX0sImVudiI6eyJlczYiOnRydWV9fX0=)) which are indeed useless escapes in template literals since they evaluate to `d` and `s`.

Your regex is actually `` /^([ds'.-_`a-z]{2,29}$)/i ``:

```js
const baseCardholderNameSurnameRegexString = `^([\d\s'.-_\`a-z]{2,29}$)`;
const cardholderRegex = new RegExp(
    baseCardholderNameSurnameRegexString,
    'i'
);

console.log(cardholderRegex); // /^([ds'.-_`a-z]{2,29}$)/i
```

If you want `\d` and `\s` in the regex, you should escape `\` in the template literal:

```js
const baseCardholderNameSurnameRegexString = `^([\\d\\s'.-_\`a-z]{2,29}$)`;
const cardholderRegex = new RegExp(
    baseCardholderNameSurnameRegexString,
    'i'
);

console.log(cardholderRegex); // /^([\d\s'.-_`a-z]{2,29}$)/i

```

**Comment by marekdon** [comment]
@mdjermanovic you are right, thank for the explanation and sorry for confusion 🤓
